pyOCD/tools/pyocd.py

```diff
@@ -218,8 +218,13 @@ def run(self):
                 else:
                     dumpHexData(data, args.read, width=args.width)
             elif args.action == ACTION_WRITE:
+                is_flash = self.isFlashWrite(args)
                 if args.width == 8:
-                    target.writeBlockMemoryUnaligned8(args.write, args.data)
+                    if is_flash:
+                        target.flash.init()
+                        target.flash.programPhrase(args.write, args.data)
```

Comment:
Good idea!  We'll definitely want this feature.  I have a couple of comments that will probably need to be addressed in the future (or now).
-You might want to check for existing data before writing.  I know on Kinetis devices if non 0xFF data is programmed to the same spot multiple times without an erase then that address can get into a state where reads cause a fault.
-Not all processors support the same sized write.  Maybe a flash attribute for minimum flash write size could be added.  Or maybe a new mode could be added to flash builder which reads the page, and then writes it back with the new data.  That way individual bytes could be programmed.
-If the user had a program running this would mess up the state of that program, since the PC would now be pointing to a breakpoint in ram.  Probably out of the scope of this pull request, but if the state of the ram and registers were recorded then the execution context could be restored.  Something like this will be necessary for flash breakpoints in the future.